src/general/WorldGenerationSettings.cs

```diff
@@ -1,11 +1,28 @@
 ﻿using System;
+using Godot;
+using Newtonsoft.Json;
 
 /// <summary>
 ///   Player configurable options for creating the game world
 /// </summary>
 public class WorldGenerationSettings
 {
-    private DifficultyPreset difficulty = null!;
+    [JsonConstructor]
+    public WorldGenerationSettings(IDifficulty difficulty)
```

Comment:
Presumably the planet settings will end up using this approach at some point too? Do we want to anticipate that now to avoid compatibility issues in future, or is that so far in the future that we can assume save breakage by that point?

Reply:
I suppose if the planet generator uses the same kind of approach where we have presets and the possibility to customize, it would need this same treatment. This needs to only be considered once the actual planet generator parameters are added, so there's nothing that needs to be done before that to prepare.